### PR TITLE
Fixed Stoploss and Take profit feature

### PR DESCRIPTION
**Pull request by Sherwyne**
resolves #

### Description

- Stop loss and take profit works fine for only trader at a time
- I temporarily removed stop_trail to easily deal with the problem in stoploss. I'll add it if there were no bugs occured

### Checklist
 - [ ] I am making a pull request from a branch other than master
 - [ ] I have read the CONTRIBUTING.md
 - [ ] I have added/edited documentation in a relevant docs/docusaurus markdown file
 - [ ] (For new docs, check if not applicable) I have added the id of a new docs md to the docs/docusaurus/sidebars.js file

### CODE REVIEW

File: python/fastquant/strategies/base.py [hunk 1]
```diff
@@ -318,76 +341,30 @@ def next(self):
...
-                        if self.transaction_logging:
-                            self.log("Stop trail: {}".format(self.stop_trail))
-                        self.sell(
-                            exectype=bt.Order.StopTrail,
-                            trailpercent=self.stop_trail,
-                            size=final_size,
-                        )
-
-                # Buy based on the opening price of the next closing day (only works "open" data exists in the dataset)
-                else:
```
Comment (enzoampil) [inline python/fastquant/strategies/base.py:356]: @Sherwyne may I ask why you deleted this whole section?

File: python/fastquant/strategies/base.py [hunk 2]
```diff
@@ -318,76 +341,30 @@ def next(self):
...
                 )
                 buy_prop_size = int(afforded_size * self.buy_prop)
+                
+                # Used for take profit method
+                self.price_bought = self.data.close[0] 
+
                 # Buy based on the closing price of the previous closing day
                 if self.execution_type == "close":
                     final_size = min(buy_prop_size, afforded_size)
-                    if self.transaction_logging:
```
Comment (enzoampil) [inline python/fastquant/strategies/base.py:324]: Why did you take out the logging here? Can we add them back?

File: python/fastquant/strategies/base.py [hunk 3]
```diff
@@ -318,76 +341,30 @@ def next(self):
...
-                        )
                         if self.transaction_logging:
                             self.log("Stop price: {}".format(stop_price))
-                        self.sell(
-                            exectype=bt.Order.Stop,
-                            price=stop_price,
-                            size=final_size,
-                        )
 
-                    if self.stop_trail:
```
Comment (enzoampil) [inline python/fastquant/strategies/base.py:382]: I just realized, can we just add back the `self.stop_trail` code block even if it's still problematic? Just so we have some version of it implemented (could be some people using it already).

**Review by enzoampil — CHANGES_REQUESTED** [review]
Hey @Sherwyne thanks for this PR! Just wanted to request the following changes:
1. Let's add back the logging code blocks - I commented on some of them asking why they were taken out
2. Let's add back the `stop_trail` code block just so we have an implementation (even if not yet robust)
3. Add back the data open based logic (I asked why you deleted the whole section)

File: python/fastquant/strategies/base.py (hunk 3 above)
Comment (Sherwyne) [reply python/fastquant/strategies/base.py:382]: Sure thing i'll add it back again

**Review by enzoampil — APPROVED** [review]
Fixed some issues with the take profit, market sizing during fractional, and formatting of the win metrics.

Automated tests all pass now as well 🍾 

You can find my extensive notebook tests in this colab [notebook](https://colab.research.google.com/drive/1mA72znQjykMCV00765V7rHNgFp2eeiX6?usp=sharing)

LGTM congrats @Sherwyne and @johnmarkacala :smile:

**Pull request merged**